spec/ddtrace/tracer_spec.rb

```diff
@@ -209,6 +278,11 @@
 
         it { expect(trace).to eq(result) }
 
+        it do
+          trace
+          expect(span.name).to eq(name)
+        end
```

Comment:
I'd still love if we could reference helpers directly. I was searching through the source to find where a `let(:span)` was defined for this context, like we do for others, and had to attach `pry` to realize that this is `TracerHelpers#span`, not a `let`.

I know I may be somewhat annoying bring this up from time to time, but I really think we should avoid if possible these module inclusions that result in methods just showing up in random places. Especially in very complex contexts (such as this one) it makes the code really hard to follow without actually firing it up in a repl/debugger and seeing what's up.

Reply:
So `tracer`, `span`, and `spans` are the only "special" ones that I think exist in `ddtrace` that keep showing up all the time.

They are always available everywhere, and I think it makes sense for them to be always available by default, given their ubiquitous utility, and so far their abstraction has held up.

Also, if never invoked, they have no side-effect, so there's no ill in having them made available to all tests.

Do you think this is something that documentation would help with? Do you think that documenting this at the top of the `spec_helper.rb`, for example, would work, given it's more or less standard to assume that `spec_helper.rb` is always(™️) loaded for an RSpec test suite.

Reply:
> So tracer, span, and spans are the only "special" ones that I think exist in ddtrace that keep showing up all the time.

If it's just me, I can go away, but I do think it makes the codebase harder to comprehend from a newer joiner's point of view (mine, lol).

> Do you think this is something that documentation would help with? Do you think that documenting this at the top of the spec_helper.rb, for example, would work, given it's more or less standard to assume that spec_helper.rb is always(™️) loaded for an RSpec test suite.

Hmm I'm not sure having it in the `spec_helper.rb` would've helped. It did not occur to me at all to search there, so if that documentation existed already, I would've completely missed it.

My suggestion would be to make it a bit more explicit in every file that uses it, e.g. something like:

```ruby
RSpec.describe SomeTracerClass do
  let(:tracer) { TracerHelpers.tracer(self) }
  let(:span) { TracerHelpers.span(self) }
  let(:spans} { TracerHelpers.spans(self) }

   # ...
end
```

or less preferable but smaller using [the use symbols as metadata option](https://relishapp.com/rspec/rspec-core/docs/helper-methods/define-helper-methods-in-a-module#use-symbols-as-metadata):

```ruby
RSpec.describe SomeTracerClass, :'enable the #tracer, #span and #spans helpers'

end
```

e.g. the key is if something shows up somewhere, there should be a hint about it in the file source code.